vignettes/rhino-fluent.Rmd

```diff
@@ -0,0 +1,321 @@
+
+Here's how you can install the `shiny.fluent` package 👇
+
+``` r
+install.packages("shiny.fluent")
+```
+
+Now that the package is installed, let us use it for building our very first `rhino` module.
+
+## Use `shiny.fluent` in your Rhino application!
+
+In Rhino, each application view is intended to live as a [Shiny module](https://shiny.rstudio.com/articles/modules.html) and use encapsulation provided by [`box` package](https://klmr.me/box/).
+Also, in rhino apps, even the base ui and server are modules!
+
+In your project, please go to the `app` folder and edit the `main.R` file with the code below
+
+``` r
+# app/main.R
+
+box::use(
+  shiny[bootstrapPage, moduleServer, NS, tags, textOutput, renderText],
+  shiny.fluent[fluentPage]
+)
+
+#' @export
+# Define user interface function
+ui <- function(id) {
+  # Create namespace for UI elements
+  ns <- NS(id)
+  # Define app layout using Fluent UI design system
+  fluentPage(
+    tags$h1(
+      textOutput(ns("message")) # Display message generated by server
+    )
+  )
+}
+
+#' @export
+# Define server logic function
+server <- function(id) {
+  # Create Shiny module
+  moduleServer(id, function(input, output, session) {
+    # Generate message to be displayed in UI
+    output$message <- renderText("👋 This text has been generated using shiny.fluent package inside a Rhino project!")
+  })
+}
+```
+
+The **`box::use`** function is used to load the necessary packages along with their required functions for the app.
+In this case, the app requires the **`shiny`** package for the basic Shiny functionalities and the **`shiny.fluent`** package for the Fluent UI design system.
+
+The **`ui`** function creates the user interface of the app.
+It takes an **`id`** argument which is used to namespace the UI elements.
+Inside the function, the **`NS`** function is called to create the namespace for the UI elements.
+The **`fluentPage`** function which is from `shiny.fluent` package is used to create the layout of the app.
+It consists of a heading (**`tags$h1`**) and a text output (**`textOutput`**) element that will display a message generated by the server.
+Fluent UI is a design system developed by Microsoft that provides a modern, responsive, and accessible user interface for web applications.
+
+The **`server`** function defines the server logic of the app.
+It takes an **`id`** argument which is used to match the namespace with the UI elements created in the **`ui`** function.
+Inside the function, the **`moduleServer`** function is called to create a Shiny module.
+The **`renderText`** function is used to generate the text output message that will be displayed in the UI.
+
+The exported **`ui`** and **`server`** functions can be used to run the app with the **`shinyApp`** function.
+The app will display a heading and a message that says "👋 This text has been generated using shiny.fluent package inside a Rhino project!" in the web browser.
```

Comment:
You describe here a general ui-server setup, so the basic Shiny. Again, this is not the goal of this tutorial.